ruby/extconf.rb

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) ZeroC, Inc. All rights reserved.
+
+require "mkmf"
+require "rbconfig"
+require 'open-uri'
+require 'fileutils'
+
+if RUBY_PLATFORM =~ /mswin|mingw/
+    puts "MinGW is not supported with Ice for Ruby."
+    exit 1
+end
+
+# On OSX & Linux bzlib.h is required.
+if not have_header("bzlib.h") then
+    puts "Bzip2 header files are missing."
+    exit 1
+end
+
+if RUBY_PLATFORM =~ /linux/
+    # On Linux openssl is required for IceSSL.
+    if not have_header("openssl/ssl.h") then
+        puts "OpenSSL header files are missing."
+        exit 1
+    end
+end
+
+$INCFLAGS << ' -Idist/ice/cpp/include'
+$INCFLAGS << ' -Idist/ice/cpp/include/generated'
+$INCFLAGS << ' -Idist/ice/cpp/src'
+$INCFLAGS << ' -Idist/ice/cpp/src/Ice/generated'
+$INCFLAGS << ' -Idist/ice/cpp/src/IceDiscovery/generated'
+$INCFLAGS << ' -Idist/ice/cpp/src/IceLocatorDiscovery/generated'
+
+$CPPFLAGS << ' -DICE_STATIC_LIBS'
+$CPPFLAGS << ' -DICE_GEM'
+
+if RUBY_PLATFORM =~ /darwin/
+    $LOCAL_LIBS << ' -lbz2 -liconv -framework Security -framework CoreFoundation'
+elsif RUBY_PLATFORM =~ /linux/
+    $LOCAL_LIBS << ' -lssl -lcrypto -lbz2 -lrt'
+end
+$CPPFLAGS << ' -w'
+$CXXFLAGS << ' -std=c++20'
```

Comment:
I believe we currently build with c++20 on macos (clang) but not on Linux (gcc).